### ISSUE TITLE
Client config does not seem to work with whonix-workstation

### ISSUE THREAD

**Issue by ddevz** [unman/qubes-print#2]
The client config does not seem to work with whonix-workstation.  By that I mean after booting, inside netstat shows listening on 9100, and .cups/client.conf points to it, but trying to print from the browser seems to do nothing.  I'm guessing a package is missing.  My first thought was cups-client may need to be installed to read .cups/client.conf but that did not seem to change anything.

**Comment by unman** [comment]
On Thu, Jul 29, 2021 at 12:36:54PM -0700, ddevz wrote:
> The client config does not seem to work with whonix-workstation.  By that I mean after booting, inside netstat shows listening on 9100, and .cups/client.conf points to it, but trying to print from the browser seems to do nothing.  I'm guessing a package is missing.  My first thought was cups-client may need to be installed to read .cups/client.conf but that did not seem to change anything.
> 

I cant help with Whonix problems, I'm afraid.
Have you confirmed you can print from a standard template based appVM?

**Comment by ddevz** [comment]
Yes.  Works great on a non-whonix system.  Thanks.  My guess is that whonix webbrowser is not reading the .cups/client.conf file.

Further research shows that it seems to be specific to torbrowser.  If you install cups-client in whonix-ws, then the "mousepad" text editor **can** print. 

I recommend you add "make sure that you have cups-client installed" to the instructions. While cups-client is normally installed by default in most distributions, the comment will really help those who do run into a distribution that does not install it by default.

**Comment by unman** [comment]
On Mon, Aug 02, 2021 at 09:59:25AM -0700, ddevz wrote:
> Yes.  Works great on a non-whonix system.  Thanks.  My guess is that whonix webbrowser is not reading the .cups/client.conf file.
> 
> Further research shows that it seems to be specific to torbrowser.  If you install cups-client in whonix-ws, then the "mousepad" text editor **can** print. 
> 
> I recommend you add "make sure that you have cups-client installed" to the instructions. While cups-client is normally installed by default in most distributions, the comment will really help those who do run into a distribution that does not install it by default.
> 

Thanks for that - I will do.

**Comment by ben-grande** [comment]
Whonix uses port 9100 for SOCKS. The port should be 631 for the IPP service, which is the port CUPS server uses by default.

Modification of `~/.cups/client.conf` or installation of `cups-client` is unnecessary if using the default port, tested with `chromium` and `mousepad` on the client.

Unman, archive this repo in favor of [shaker/print](https://github.com/unman/shaker/tree/main/print) to avoid code duplication?

Are you accepting contributions? I did propose changes to [cacher](https://github.com/unman/shaker/issues/30#issuecomment-1986864647) and there was no reply.
